### ISSUE TITLE
 Error: Could not encrypt data for service RN_KEYCHAIN_DEFAULT_ALIAS

### ISSUE THREAD

**Issue by domandtom-andinieves** [oblador/react-native-keychain#157]
Keychain is working perfectly fine in iOS, but no luck in Android. I am totally blocked by this. What seems the problem with my code? Can you help me guys?

 `Error: Could not encrypt data for service RN_KEYCHAIN_DEFAULT_ALIAS`

```
const allValues = await readKeychain();
try{
await Keychain.setGenericPassword('mykey','value', {
        accessible: Keychain.ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
      });
}catch(e){
// I logged it here
}

**Comment by pscarey** [comment]
We've seen this issue on a Sony Xperia, Running Android 8.0.0. It appears device specific at this stage. Other Android users with a range of devices are fine.

Issue appears to be [here](https://github.com/oblador/react-native-keychain/blob/8f33c0f75032aa7309e4cb574299fc4785076927/android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageKeystoreAESCBC.java#L138-L154).

Unfortunately, our error tracking isn't passing the *actual* error which is thrown, just the `CryptoFailedException` with a generic description.

**Comment by jinshin1013** [comment]
Can anything be done for this? It's happening on Nexus 5X - 8.1.0 emulator as well.

**Comment by tomasmedina** [comment]
I just started to have this problem. I was running 8.1.0 on a Nexus 5X on the emulator and out of nowhere this error popped up. It would be nice to have an update on this error...

**Comment by zigzactly** [comment]
I am having this same issue on Nexus 5X emulator 8.1.0.

**Comment by jamesmcn1** [comment]
Same issue on Nexus 5X emulator (Android 8.1.0), running react-native 0.57.8 with react-native-keychain 3.0.0. Any fixes for this yet?

**Comment by vonovak** [comment]
There are now 6 people having this issue. If you can reproduce the issue, you can also fix it :) It would be great if someone could step up, fix it and submit a PR. Thanks.

**Comment by vniche** [comment]
So, i don't know if it is exactly related since my code differs from the snippet @dnt-andinieves posted but i was getting this when password/value content got bigger, i can't say exactly at which point, but a content shortening made the error "disappear".

**Comment by chaudhryjunaid** [comment]
The issue is actually here: https://github.com/oblador/react-native-keychain/blob/8f33c0f75032aa7309e4cb574299fc4785076927/android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageKeystoreAESCBC.java#L74
@pscarey

**Comment by chaudhryjunaid** [comment]
👍  bump!

**Comment by lucaashrq** [comment]
Any update for this issue?

**Comment by tbrent** [comment]
Same issue with Pixel 2 API 28, Android 9.0, react-native 0.61.4, react-native-keychain 6.0.0.

**Comment by 1jedrzej1** [comment]
One of possibility is not enought storage space on emulator or phone. Please checks that.
